src/routes/DigitalAssets/modules/digitalAssets.js

```diff
@@ -1,299 +1,139 @@
-  type: string,
-  payload: {
-    currentAddress: ?Address,
-  },
-} => ({
-  type: SET_CURRENT,
-  payload: {
-    currentAddress,
-  },
-})
-
-export const getBalances = (): { type: string } => ({
-  type: GET_BALANCES,
-})
-
-export const getBalancesSuccess = (balances: Balances): {
-  type: string,
-  payload: {
-    balances: Balances,
-  },
-} => ({
-  type: GET_BALANCES_SUCCESS,
-  payload: {
-    balances,
-  },
-})
-
-export const getBalancesError = (err: Object): {
-  type: string,
-  payload: Object,
-  error: boolean,
-} => ({
-  type: GET_BALANCES_ERROR,
-  payload: err,
-  error: true,
-})
-
-export const setBalancesLoading = (isBalancesLoading: boolean): {
-  type: string,
-  payload: {
-    isBalancesLoading: boolean,
-  },
-} => ({
-  type: SET_BALANCES_LOADING,
-  payload: {
-    isBalancesLoading,
-  },
-})
-
-export const setBalanceByAddress = (address: Address, balance: number): {
-  type: string,
-  payload: {
-    address: Address,
-    balance: number,
-  },
-} => ({
-  type: SET_BALANCE_BY_ADDRESS,
-  payload: {
-    address,
-    balance,
-  },
-})
+export const SET_INITIAL_ITEMS = '@@digitalAssets/SET_INITIAL_ITEMS'
+export const ADD_CUSTOM_ASSET = '@@digitalAssets/ADD_CUSTOM_ASSET'
+export const REMOVE_ASSET = '@@digitalAssets/REMOVE_ASSET'
+export const UPDATE_ASSET = '@@digitalAssets/UPDATE_ASSET'
+
+export function setInitialItems(items: DigitalAssets) {
+  return {
+    type: SET_INITIAL_ITEMS,
+    payload: {
+      items,
+    },
+  }
+}
 
-export const setBalanceByAddressSuccess = (balances: Balances): {
-  type: string,
-  payload: {
-    balances: Balances,
-  },
-} => ({
-  type: SET_BALANCE_BY_ADDRESS_SUCCESS,
-  payload: {
-    balances,
-  },
-})
+export function addCustomAsset(address: Address, name: string, symbol: string, decimals: number) {
+  return {
+    type: ADD_CUSTOM_ASSET,
+    payload: {
+      address,
+      name,
+      symbol,
+      decimals,
+    },
+  }
+}
 
-export const search = (searchQuery: string): {
-  type: string,
-  payload: {
-    searchQuery: string,
-  },
-} => ({
-  type: SEARCH,
-  payload: {
-    searchQuery,
-  },
-})
+export function removeAsset(assetAddress: Address) {
+  return {
+    type: REMOVE_ASSET,
+    payload: {
+      address: assetAddress,
+    },
+  }
+}
 
-export const searchSuccess = (foundAssets: Addresses): {
-  type: string,
-  payload: {
-    foundAssets: Addresses,
-  },
-} => ({
-  type: SEARCH_SUCCESS,
-  payload: {
-    foundAssets,
-  },
-})
+export function updateAsset(address: Address, name: string, symbol: string, decimals: number) {
+  return {
+    type: UPDATE_ASSET,
+    payload: {
+      address,
+      name,
+      symbol,
+      decimals,
+    },
+  }
+}
 
-export const searchError = (err: Object): {
-  type: string,
-  payload: Object,
-  error: boolean,
-} => ({
-  type: SEARCH_ERROR,
-  payload: err,
-  error: true,
-})
+export type DigitalAssetsActions = ExtractReturn<typeof setInitialItems>
```

Comment:
actually it is one action :)
just enum type, so better to use `DigitalAssetsAction`, I suppose

Reply:
It checked inside saga, then UPDATE_ASSET event sent